lib/agent/cloud/common.go

```diff
@@ -76,20 +83,33 @@ func UpdateRateLimitingConfig() {
 		}
 
 		log.Infof("Got new rate limiting endpoint config and storing to map: %v", newEndpointConfig)
-		globals.RateLimitingMap[k] = &RateLimitingValue{
+		rateLimitingValue := &RateLimitingValue{
 			Config: RateLimitingConfig{
 				MaxRequests:         newEndpointConfig.RateLimiting.MaxRequests,
 				WindowSizeInMinutes: newEndpointConfig.RateLimiting.WindowSizeInMS / MinRateLimitingIntervalInMs},
 			UserCounts: make(map[string]*RateLimitingCounts),
 			IpCounts:   make(map[string]*RateLimitingCounts),
 		}
+
+		if isWildcardEndpoint(k.Route) {
+			routeRegex, err := regexp.Compile(strings.ReplaceAll(k.Route, "*", ".*"))
```

Comment:
For uniformity, I know it's just a capture group, but makes it easier to debug if route string is not all good
```suggestion
			routeRegex, err := regexp.Compile(strings.ReplaceAll(k.Route, "*", "(.*)"))
```